src/clr-addons/numericfield/numeric-field.ts

```diff
@@ -120,11 +122,14 @@ export class ClrNumericField implements OnInit, OnDestroy, AfterViewChecked {
           if (event.keyCode === BACK_KEYCODE && value.substr(cursorStart - 1, 1) === this.groupingSeparator) {
             event.target.value = value.substring(0, cursorStart - 2) + value.substring(cursorStart - 1, value.length);
             event.target.selectionStart = event.target.selectionEnd = cursorStart - 1;
+            return false;
           } else if (event.keyCode === DEL_KEYCODE && value.substr(cursorStart, 1) === this.groupingSeparator) {
             event.target.value = value.substring(0, cursorStart + 1) + value.substring(cursorStart + 2, value.length);
             event.target.selectionStart = event.target.selectionEnd = cursorStart + 1;
+            return false;
           }
         }
+        this.originalValue = NaN;
```

Comment:
I guess you did the return false, because of this line. But is this line needed?
The value will get emitted anyways as "userEntered" will be true.

Reply:
might be unnecessary, but this also more clearly indicates a 'dirty' input field